corehq/motech/repeaters/expression/repeaters.py

```diff
@@ -222,39 +222,37 @@ def send_request(self, repeat_record, payload):
         response = super().send_request(repeat_record, payload)
         if is_success_response(response) and 'error' in response.json():
             # It _looks_ like a success response, but it's an error. :/
-            return self._error_response(response.json())
+            return self._error_response(response.json()['error'])
         return response
 
     @staticmethod
-    def _error_response(response_json):
+    def _error_response(error_json):
         """
         The ArcGIS API returns error responses with status code 200.
 
         This method extracts the details from the response JSON, and
         returns a RepeaterResponse with the error details so that the
         response will be handled correctly.
 
-        >>> response_json = {
-        ...     "error": {
-        ...         "code": 503,
-        ...         "details": [],
-        ...         "message": "An error occurred."
-        ...     }
+        >>> error_json = {
+        ...     "code": 503,
+        ...     "details": [],
+        ...     "message": "An error occurred."
         ... }
-        >>> resp = ArcGISFormExpressionRepeater._error_response(response_json)
+        >>> resp = ArcGISFormExpressionRepeater._error_response(error_json)
         >>> resp.status_code
         503
         >>> resp.reason
         'An error occurred.'
 
         """
-        reason = response_json['error']['message']
-        if 'messageCode' in response_json['error']:
-            reason += f' ({response_json["error"]["messageCode"]})'
+        reason = error_json.get('message', 'Unspecified ArcGIS REST API error')
```

Comment:
should we add a message if none if provided?
(I assume RepeaterResponse is what we store and show on HQ as the response received from the external API)
I'd expect HQ to tell what response was received from the API, but if we overwrite that, that could cause confusion.

same for the code below.

Reply:
I see your comment here now, https://github.com/dimagi/commcare-hq/pull/35217#discussion_r1800196190
So overriting `code` seems okay. Though if we can add a comment for it in the code, that would be helpful.